sources/lib/reasoners/ccx.ml

```diff
@@ -467,7 +467,10 @@ module Main : S = struct
       let nuse = Use.up_add env.use t rt lvs in
 
       (* If finitetest is used we add the term to the relation *)
-      let rel = Rel.add env.relation nuf rt t in
+      let rel,eqs = Rel.add env.relation nuf rt t in
+      Debug.make_cst "relation" t ctx;
+      (* We add terms made from relations as fact *)
+      List.iter (fun (a,ex) -> add_fact facts (LTerm a, ex, Th_util.Other)) eqs;
```

Comment:
I don't think those changes directly relate to exponentiation, what are they for ?

Reply:
They are not directly related of exponentiation but we needed to add a list of terms deduce by the add function. I should have made 2 commits.